stripe/src/main/java/com/stripe/android/net/StripeApiHandler.java

```diff
@@ -419,11 +478,15 @@ static String createQuery(Map<String, Object> params)
         JSONObject headerMappingObject = new JSONObject(propertyMap);
         headers.put("X-Stripe-Client-User-Agent", headerMappingObject.toString());
 
-        if (apiVersion != null) {
-            headers.put("Stripe-Version", apiVersion);
+        if (options != null && options.getApiVersion() != null) {
+            headers.put("Stripe-Version", options.getApiVersion());
         }
 
-        if (options.getIdempotencyKey() != null) {
+        if (options != null && options.getGuid() != null && !TextUtils.isEmpty(options.getGuid())) {
+            headers.put("Cookie", "m=" + options.getGuid());
```

Comment:
should we be using the same method to generate headers for calls to `m.stripe.com`? seems like it'd be safer to separate the two, since they have different requirements